File: C++/83_RemoveDuplicatesfromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (head == NULL) {
            return head;
        }
        
        ListNode* p = head;
        ListNode* q = head->next;
        while (q != NULL) {
            if (p->val == q->val) {
                p->next = q->next;
            } else {
                p = p->next;
            }
            q = q->next;
        }
        
        return head;
    }
};

File: Golang/83_RemoveDuplicatesfromSortedList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil {
        return head
    }
    
    p := head
    q := head.Next
    for {
        if q == nil {
            break
        }
        
        if p.Val == q.Val {
            p.Next = q.Next
        } else {
            p = p.Next
        }
        q = q.Next
    }
    return head
}
